include/onnxruntime/core/framework/op_kernel.h

```diff
@@ -129,6 +129,12 @@ class OpKernel {
     return Status::OK();
   }
 
+  // Override this function to handle the fastmath implementation mode of the kernel
+  // the default behavior is ignore this request
+  virtual Status SetFastMathMode(bool /*enable = true*/) {
+    return Status::OK();
+  }
```

Comment:
I think this is way too specific to be adding to the base OpKernel class, and the virtual method is going to impact binary size. We need a more generic way to make these extremely specific config values available to the kernel. 

One option would be to add the SessionOptions to the CPU EP (we do this for the [XNNPACK EP](https://github.com/microsoft/onnxruntime/blob/6d7519ede8298a422e84e70bfdf01cc46fbf76c3/onnxruntime/core/providers/xnnpack/xnnpack_provider_factory.cc#L13)). The EP instance should be accessible via OpKernelInfo.GetExecutionProvider() in the MatMul ctor, so if CPUExecutionProvider exposes a way to read config values the kernel can access them. 

We can look for other options as well.

Reply:
Alternatively adding a helper to OpKernelInfo that allows reading a config option might be slightly cleaner/simpler. We will need to discuss and agree upon the best approach which may require waiting until some folks that are currently on vacation return.

e.g.

```c++
class OpKernelInfo : public OpNodeProtoHelper<ProtoHelperNodeContext> {
 public:
  explicit OpKernelInfo(const onnxruntime::Node& node,
                        const KernelDef& kernel_def,
                        const IExecutionProvider& execution_provider,
                        const std::unordered_map<int, OrtValue>& constant_initialized_tensors,
                        const OrtValueNameIdxMap& mlvalue_name_idx_map,
                        const DataTransferManager& data_transfer_mgr,
                        const ConfigOptions& config_options,
                        const AllocatorMap& allocators = {});
  ...
  // call GetConfigEntry instead of having a basic getter for ConfigOptions so we don't need to add the entire ConfigOptions class to the provider bridge
  std::optional<std::string> GetConfigOption(const std::string& key) const {
    return config_options_.GetConfigEntry(key);
  }
```